portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/Endpoints/EndpointOverview.jsx

```diff
@@ -564,7 +564,7 @@ function EndpointOverview(props) {
         setIsCustomBackendSelected(false);
         const selectedKey = typeChangeConfirmation.type || value;
         if (selectedKey === 'INLINE' || selectedKey === 'MOCKED_OAS') {
-            const tmpConfig = createEndpointConfig('prototyped');
+            const tmpConfig = createEndpointConfig(selectedKey);
```

Comment:
Changing the endpoint type is currently working as intended. If you select the “Mock Implementation” radio button after choosing another endpoint type, the UI correctly navigates to the correct view.